File: api/src/main/java/com/ruby/rubia_server/core/entity/CompanyGroup.java
// src/main/java/com/ruby/rubia_server/core/entity/CompanyGroup.java
package com.ruby.rubia_server.core.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "company_groups")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CompanyGroup {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true)
    private String name; // Nome do grupo (ex: "Grupo GSH")

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "is_active")
    @Builder.Default
    private Boolean isActive = true;

    // Relação com as empresas que fazem parte deste grupo
    @OneToMany(mappedBy = "companyGroup", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<Company> companies;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: api-dotnet/Rubia.Server/Entities/CompanyGroup.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Rubia.Server.Entities;

[Table("company_groups")]
public class CompanyGroup : BaseEntity
{
    [Required]
    [Column("name")]
    public string Name { get; set; } = string.Empty; // Nome do grupo (ex: "Grupo GSH")

    [Column("description", TypeName = "TEXT")]
    public string? Description { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    // Relação com as empresas que fazem parte deste grupo
    public virtual ICollection<Company> Companies { get; set; } = new List<Company>();
}